Doc/library/warnings.rst

```diff
@@ -522,6 +522,52 @@ Available Functions
    and calls to :func:`simplefilter`.
 
 
+.. decorator:: deprecated(msg, *, category=DeprecationWarning, stacklevel=1)
+
+   Decorator to indicate that a class, function or overload is deprecated.
+
+   Usage::
+
+      from warnings import deprecated
+      from typing import overload
+
+      @deprecated("Use B instead")
+      class A:
+          pass
+
+      @deprecated("Use g instead")
+      def f():
+          pass
+
+      @overload
+      @deprecated("int support is deprecated")
+      def g(x: int) -> int: ...
+      @overload
+      def g(x: str) -> int: ...
+
+   When this decorator is applied to an object,
+   :term:`static type checkers <static type checker>`
+   will generate a diagnostic on usage of the deprecated object.
+
+   The warning specified by ``category`` will be emitted on use
+   of deprecated objects. For functions, that happens on calls;
+   for classes, on instantiation. If the ``category`` is ``None``,
+   no warning is emitted. The ``stacklevel`` determines where the
+   warning is emitted. If it is ``1`` (the default), the warning
+   is emitted at the direct caller of the deprecated object; if it
+   is higher, it is emitted further up the stack.
+
+   The decorator sets the ``__deprecated__``
+   attribute on the decorated object to the deprecation message
+   passed to the decorator. If applied to an overload, the decorator
```

Comment:
This is quite a long sentence, I had to read it a couple of times. Perhaps this might help clarify? Or can it be reworded or split?

```suggestion
   The decorator sets the ``__deprecated__``
   attribute on the decorated object to the deprecation message that was
   passed to the decorator. If applied to an overload, the decorator
```

Reply:
Much better, thanks!